src/components/FiltersButton/FiltersButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { useState } from 'react';
+import commonOptionButtonStyles from '../../common/commonOptionButton.module.css';
+import styles from './FiltersButton.module.css';
+
+const filterOptions: string[] = [
+   'Animals',
+   'Health',
+   'Education',
+   'Environment',
+];
+
+function FiltersButton() {
+   const [selectedFilters, setSelectedFilters] = useState<string[]>([]);
+   const [isOpen, setIsOpen] = useState<boolean>(false);
+
+   const toggleIsOpen = () => {
+      setIsOpen(!isOpen);
+   };
+
+   const handleFilterClick = (filter: string) => {
+      if (selectedFilters.includes(filter)) {
+         setSelectedFilters(selectedFilters.filter((f) => f !== filter));
+      } else {
+         setSelectedFilters([...selectedFilters, filter]);
```

Comment:
It is sadly common to use ```f``` and such as variables, but lets call it something. For example, ```selectedFilter``` or ```currentFilter``` and so on.